client/src/components/profile/components/portfolio.tsx

```diff
@@ -55,95 +49,93 @@ function createFindById(id: string) {
   return (p: PortfolioProjectData) => p.id === id;
 }
 
-class PortfolioSettings extends Component<PortfolioProps, PortfolioState> {
-  validationImage: HTMLImageElement;
-  static displayName: string;
-  constructor(props: PortfolioProps) {
-    super(props);
-    this.validationImage = new Image();
-    const { portfolio = [] } = props;
-
-    this.state = {
-      portfolio: [...portfolio],
-      unsavedItemId: null,
-      isImageValid: {
-        state: 'success',
-        message: ''
-      }
-    };
-  }
-
-  toggleEditing = () => {
-    this.props.setIsEditing(false);
+const PortfolioSettings: React.FC<PortfolioProps> = props => {
+  const {
+    t,
+    portfolio: initialPortfolio = [],
+    setIsEditing,
+    updatePortfolio
+  } = props;
+  const [portfolio, setPortfolio] = useState([...initialPortfolio]);
```

Comment:
```suggestion
  const [portfolio, setPortfolio] = useState(initialPortfolio);
```
We don't need to be defensive. As long as we don't mutate `initialPortfolio` (we don't) this is fine.